frontend/src/views/Queue.vue

```diff
@@ -0,0 +1,219 @@
+<template>
+  <div class="Queue">
+    <h1>Welcome to the Queue page</h1>
+    <br>
+      <h2>Your UserID is : {{uid}}</h2>
+    <br>
+      <h2>You are in Store : {{storeId}}</h2>
+    <div v-if="isEnabled">
+      <div v-if="isUserLoaded">
+        <div v-if="!isUserEnrolled">
+          <h2>The size of the queue is : {{queueLength}}</h2>
+        </div>
+        <div v-else>
+          <h2>Your position in the queue is : {{queuePosition}}</h2>
+          <h2>Your token number is : {{tokenNumber}}</h2>
+        </div>
+        <button :disabled="isUserEnrolled" @click="enterQueue">Enter Queue</button>
+        <br><br>
+        <button :disabled="!isUserEnrolled" @click="exitQueue">Leave Queue</button>
+      </div>
+    </div>
+    <div v-else>
+      <h2>Queue Disabled.</h2>
+    </div>
+    <br><br>
+    <p><router-link :to="{ name: 'Home' }">Back</router-link></p>
+  </div>
+</template>
+
+<script>
+import firebase from "firebase";
+import { database_call } from "../database.js";
+import { waiting_time } from "../waitingtime.js";
+// @ is an alias to /src
+
+export default {
+  name: "Queue",
+  components: {},
+  data(){
+    return{
+      storeId: this.$route.params.StoreId,
+      uid: null,
+      currentUserKey: null,
+      currentStoreKey: null,
+      isUserLoaded: false,
+      isUserEnrolled: false,
+      queueLength: 0,
+      queuePosition: 0,
+      waitingTime: 0,
+      tokenNumber: null,
+      isEnabled: null
+    }
+  },
+  methods: {
+    logout: function() {
+      firebase
+        .auth()
+        .signOut()
+        .then(() => {
+          this.$router.replace("login");
+        });
+    },
+    storeInit: function(){
+      let dbRef = firebase.database().ref();
+      var that = this;
+      dbRef.child("Store/"+ this.storeId + "/IsEnabled").once("value", snap=>{
+        that.isEnabled = snap.val();
+      })
+    },
+    userInit: function() {
+      //Function to populate intial data values
+      let dbRef = firebase.database().ref();
```

Comment:
you should not create dbRef anywhere else. Only database.js file should do that. From everywhere else, call the methods of database.js